help_to_heat/frontdoor/interface.py

```diff
@@ -613,9 +616,40 @@ def get_address_and_epc_lmk(self, building_name_or_number, postcode):
 
     def get_epc(self, lmk):
         epc_api = EPCApi()
-        epc_details = epc_api.get_epc_details(lmk)
-        epc_recommendations = epc_api.get_epc_recommendations(lmk)
-        return epc_details, epc_recommendations
+        epc_details_response = self._get_epc_details(epc_api, lmk)
+        epc_recommendations = self._get_epc_recommendations(epc_api, lmk)
+        return (
+            epc_details_response,
+            epc_recommendations,
+        )
+
+    def _get_epc_details(self, epc_api, lmk):
+        return epc_api.get_epc_details(lmk)["rows"][0]
+
+    RECOMMENDATION_RETRIES_COUNT = 3
+
+    def _get_epc_recommendations(self, epc_api, lmk):
+        for i in range(self.RECOMMENDATION_RETRIES_COUNT):
+            try:
+                return epc_api.get_epc_recommendations(lmk)["rows"]
+            except RequestException as requestException:
+                match requestException.response.status_code:
+                    case 500:
+                        # if on the last try
+                        if i == self.RECOMMENDATION_RETRIES_COUNT - 1:
+                            # raise this to logs, though let user continue with no recommendations
+                            logger.exception(requestException)
+                            return []
+
+                        # else if not on last try, try again
+                        continue
```

Comment:
I think we should have some exponential backoff here - there's a good article [here](https://encore.dev/blog/retries) which explains why it's a good idea. It's probably not a strict necessity at this scale, and I don't think we need to worry about jitter, but it's good practice and also gives us some more resilience against short-term problems with the API.

Reply:
have added some